packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -250,6 +250,25 @@ abstract class Target {
   }
 }
 
+class CompositeTarget extends Target {
```

Comment:
I think there is a composite target class in the command file, could you delete that one and move it here. 

Also worth documenting a bit, at least for your own use case.